From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Anselmo L. S. Melo" <anselmo.melo@intel.com>
Date: Thu, 20 Jun 2019 19:38:13 -0700
Subject: [PATCH] build: Add configure option to use system-libpng

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 configure.ac | 47 +++++++++++++++++++++++++++++++++--------------
 1 file changed, 33 insertions(+), 14 deletions(-)

diff --git a/configure.ac b/configure.ac
index 698abd3..8e64a5d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1240,21 +1240,40 @@ else
   LIBPNGDIR=src
   PNGDEVS=''
   PNGDEVS_ALL='png48 png16m pnggray pngmono pngmonod png256 png16 pngalpha'
-  AC_MSG_CHECKING([for local png library source])
-  if test -f $srcdir/libpng/pngread.c; then
-          AC_MSG_RESULT([yes])
-          SHARE_LIBPNG=0
-          LIBPNGDIR=$srcdir/libpng
+  AC_ARG_WITH([system-libpng], AC_HELP_STRING([--with-system-libpng],
+                                               [Force using the systems libpng]),
+              [], [with_system_libpng=no])
+  case "x$with_system_libpng" in
+    xyes)
+      if test "x$PKGCONFIG" != x; then
+        AC_MSG_CHECKING(for libpng  with pkg-config)
+        if $PKGCONFIG --exists libpng; then
+          AC_MSG_RESULT(yes)
+          CFLAGS="$CFLAGS `$PKGCONFIG --cflags libpng`"
+          LIBS="$LIBS `$PKGCONFIG --libs libpng`"
+          HAVE_SYSTEM_LIBPNG=1
+          SHARE_LIBPNG=1
           PNGDEVS="$PNGDEVS_ALL"
-  else
-          AC_MSG_RESULT([no])
-          AC_CHECK_LIB(png, png_create_write_struct, [
-            AC_CHECK_HEADERS(png.h, [
-                  SHARE_LIBPNG=1
-                  PNGDEVS="$PNGDEVS_ALL"
-            ], [SHARE_LIBPNG=0])
-          ], [SHARE_LIBPNG=0], [-lz])
-  fi
+        fi
+      fi
+      ;;
+    xno)
+      AC_MSG_CHECKING([for local png library source])
+      if test -f $srcdir/libpng/pngread.c; then
+              AC_MSG_RESULT([yes])
+              SHARE_LIBPNG=0
+              LIBPNGDIR=$srcdir/libpng
+              PNGDEVS="$PNGDEVS_ALL"
+      else
+              AC_MSG_RESULT([no])
+              AC_CHECK_LIB(png, png_create_write_struct, [
+                AC_CHECK_HEADERS(png.h, [
+                      SHARE_LIBPNG=1
+                      PNGDEVS="$PNGDEVS_ALL"
+                ], [SHARE_LIBPNG=0])
+              ], [SHARE_LIBPNG=0], [-lz])
+      fi
+  esac
   if test -z "$PNGDEVS"; then
     AC_MSG_NOTICE([disabling png output devices])
   fi
